### ISSUE TITLE
undefined reference to 'check_srtp_roc' when PJSUA_MEDIA_HAS_PJMEDIA = 0

### ISSUE THREAD

**Issue by johado** [pjsip/pjproject#3067]
### Describe the bug

We're using gstreamer as media backend so we have #define PJSUA_MEDIA_HAS_PJMEDIA 0
With 2.12 it no longer compiles/links since the #if statements in src/pjsua-lib/pjsua_media.c removes the check_srtp_roc() function.

### Steps to reproduce

Add
#define PJSUA_MEDIA_HAS_PJMEDIA 0 
in config_site.h and try to compile.

### PJSIP version

2.12

### Context

Linux with #define PJSUA_MEDIA_HAS_PJMEDIA 0 in config_site.h

### Log, call stack, etc

```shell
/path/recipe-sysroot-native/usr/bin/arm-poky-linux-gnueabi/../../libexec/arm-poky-linux-gnueabi/gcc/arm-poky-linux-gnueabi/11.2.0/ld: /tmp/cc3CqFzc.ltrans3.ltrans.o: in function `pjsua_media_channel_update':
| /path/../../../pjproject/2.12-r0/pjproject-2.12/pjsip/build/../src/pjsua-lib/pjsua_media.c:3902: undefined reference to `check_srtp_roc'
| collect2: error: ld returned 1 exit status
```

**Comment by johado** [comment]
Added https://github.com/pjsip/pjproject/pull/3072

**Comment by sauwming** [comment]
Fixed by #3019.
